src/recipes/index.js

```diff
@@ -30,8 +30,46 @@ function getRandom() {
   return rand;
 }
 
+function getSimilarRecipe(id) {
+  const { keywords, ingredients, name } = getRecipe(id);
+  const similarities = [];
+  recipes.forEach((recipe) => {
```

Comment:
What are your thoughts on doing an async/parallel iteration on `recipes`? Via `Promise.all` or `async.forEach`?

Reply:
hmmm I haven't considered those yet. How would you implement an async iteration for the foreach loop?

Reply:
Normally I would use something like `async`'s `.each`:
http://caolan.github.io/async/v3/docs.html#each

But I believe `Promise.all` does the same. However, I have not used it yet:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all

This way we can iterate the long list of recipes in parallel without blocking, and once they are resolved, we can analyze the array of items.

Reply:
I'd be interesting to see a benchmark for comparison.

Reply:
my initial assumption is that iterating over json recipes are a sync action, that is, it's always certain that we're able to retrieve them. That, in addition to the fact that node is single-threaded, lead me to think the best we can do is just synchronously iterating the recipes.

 Tasks such as requesting over a third party API or anything with Buffers where the request-response duration is uncertain would benefit from Promise.all/async.forEach approach.

 Refer to this [thread](https://codereview.stackexchange.com/questions/147455/parallel-and-sequential-array-looping-with-async-await). As of now, I implemented the promise.all approach.

Reply:
That makes sense to use when the response time is uncertain. Thanks for your effort on this.